### DISCUSSION TITLE
Renaming Python "master" branch to "main" (1.4k PRs & 700 repositories) triggered server HTTP error 500

### DISCUSSION THREAD

**Discussion by vstinner** [community/community#21636] [New to GitHub]
<p>Hi, the default “master” branch of the python/cpython project was renamed to “main”. The dialog announced that 1.4k pull requests and 747 repositories will be impacted. A few minutes later, a “main” branch was created, the default remains “master”, and <a href="https://github.com/python/cpython/branches" rel="noopener nofollow ugc">https://github.com/python/cpython/branches</a> displays an HTTP error 500.</p>
<p>Right now, <a href="https://github.com/python/cpython/pulls/" class="inline-onebox" rel="noopener nofollow ugc">Pull requests · python/cpython · GitHub</a> PRs still point to “master” (I checked a few of them, there are 1.4k+ open PRs, 25k+ PRs in total).</p>
<p>Should we retry to rename the branch? Is a job still running in the GitHub server side?</p>

**Comment by vstinner** [comment]
<p>Copy of the “rename branch” dialog:</p>
<blockquote>
<p>Rename master to:<br>
main</p>
<p>Renaming the branch:</p>
<ul>
<li>Will update 1.4k pull requests targeting this branch accross 747 repositories.</li>
<li>Will update 1 branch protection that explicitly targets master.</li>
<li>Will not update your members’ local environments. (Learn more).</li>
</ul>
</blockquote>
<p>Screenshot: <a href="https://twitter.com/DeejayNof/status/1389328007375036421/photo/1" rel="noopener nofollow ugc">https://twitter.com/DeejayNof/status/1389328007375036421/photo/1</a></p>

**Comment by jcgoble3** [comment]
<p>Looks like it just needed time to process. <a href="https://github.com/python/cpython/branches" rel="noopener nofollow ugc">https://github.com/python/cpython/branches</a> now shows “main” as the default and no sign of a “master” branch. I checked one random PR and it now targets “main”.</p>

**Comment by ambv** [comment]
<p>Not quite: <a href="https://twitter.com/willsmythe/status/1389332999720800256" rel="noopener nofollow ugc">https://twitter.com/willsmythe/status/1389332999720800256</a></p>

**Comment by aharpole** [comment, marked as answer]
<p>Hi, I’m an engineer on GitHub’s coding team. I just manually processed that branch rename. One of the initial steps of the rename was taking long enough that it timed out and so the rename didn’t finish on its own.</p>
<p>Generally you can expect a rename to be finished within a couple minutes of when it starts, but if you do find one that hangs like this feel free to reach out to support and we can take a look.</p>

**Comment by gpshead** [comment]
<p>Thanks!  We suspected it was something like that.</p>
